src/Service/Api.php

```diff
@@ -1811,4 +1822,92 @@ public function getAutoscalingSettings(Environment $environment)
         }
         return $settings;
     }
+
+    /**
+     * Loads the next environment deployment and caches it statically.
+     *
+     * The static cache means it can be reused while running a sub-command.
+     *
+     * @param Environment $environment
+     * @param bool $reset
+     * @return EnvironmentDeployment
+     */
+    public function loadNextDeployment(Environment $environment, $reset = false)
+    {
+        $cacheKey = $environment->project . ':' . $environment->id;
+        if (isset(self::$cachedNextDeployment[$cacheKey]) && !$reset) {
+            return self::$cachedNextDeployment[$cacheKey];
+        }
+        $progress = new ProgressMessage($this->stdErr);
+        try {
+            $progress->show('Loading deployment information...');
+            $next = $environment->getNextDeployment();
+            if (!$next) {
+                throw new EnvironmentStateException('No next deployment found', $environment);
+            }
+        } finally {
+            $progress->done();
+        }
+        return self::$cachedNextDeployment[$cacheKey] = $next;
+    }
+
+    /**
+     * Lists services in a deployment.
+     *
+     * @param EnvironmentDeployment $deployment
+     *
+     * @return array<string, WebApp||Worker|Service>
+     *     An array of services keyed by the service name.
+     */
+    public function allServices(EnvironmentDeployment $deployment)
+    {
+        $webapps = $deployment->webapps;
+        $workers = $deployment->workers;
+        $services = $deployment->services;
+        ksort($webapps, SORT_STRING|SORT_FLAG_CASE);
+        ksort($workers, SORT_STRING|SORT_FLAG_CASE);
+        ksort($services, SORT_STRING|SORT_FLAG_CASE);
+        return array_merge($webapps, $workers, $services);
+    }
+
+    /**
+     * Check if project supports guaranteed resources.
+     *
+     * @param array $projectInfo
+     *
+     * @return bool
+     *  True if guaranteed CPU is supported, false otherwise.
+     */
+    public function supportsGuaranteedCPU(array $projectInfo)
+    {
+        return !empty($projectInfo["settings"]["enable_guaranteed_resources"]) &&
+            !empty($projectInfo["capabilities"]["guaranteed_resources"]["enabled"]);
+    }
+
+    /**
+     * Check if environment has guaranteed CPU.
+     *
+     * @param \Platformsh\Client\Model\Environment $environment
+     *
+     * @return bool
+     */
+    public function environmentHasGuaranteedCPU(Environment $environment)
+    {
+        $nextDeployment = $this->loadNextDeployment($environment);
```

Comment:
Hm. Does this need to be based on the `next` deployment, not on the `current` one?


```suggestion
        $deployment = $this->getCurrentDeployment($environment);
```

Reply:
Yeah, that makes sense for activate and branch operations.